### PR TITLE
Add error codes

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth

### CODE REVIEW

File: src/librustc_borrowck/diagnostics.rs [hunk 1]
```diff
@@ -263,12 +263,46 @@ fn mutable() {
...
+
+```
+let mut i = 0;
+let mut x = &mut i;
+let mut a = &mut i;
+// error: cannot borrow `i` as mutable more than once at a time
+```
+
+Please note that in rust, you can have as many reference on a variable as you
+want, but only one can be mutable. Example:
```
Comment (Manishearth) [inline src/librustc_borrowck/diagnostics.rs]: You can either have many immutable references, or one mutable reference.

(You cannot have an immutable reference + a mutable reference at any time)

Link to the book on ownership and borrowing for more info

**Comment by GuillaumeGomez** [comment]
Updated !

**Comment by GuillaumeGomez** [comment]
I'll stop this PR here and open a new one if I find other error codes not filed.

File: src/librustc_borrowck/diagnostics.rs [hunk 2]
```diff
@@ -263,12 +263,50 @@ fn mutable() {
...
+
+```
+let mut i = 0;
+let mut x = &mut i; // ok!
+
+// or:
+let mut i = 0;
+let a = &i; // ok!
+let b = &i; // still ok!
+let c = &i; // super still ok!
```
Comment (Manishearth) [inline src/librustc_borrowck/diagnostics.rs]: no "super" :smile:

Comment (GuillaumeGomez) [inline src/librustc_borrowck/diagnostics.rs]: You sure ? :(

Comment (CruzBishop) [inline src/librustc_borrowck/diagnostics.rs]: He's probably super sure :cry:

Comment (GuillaumeGomez) [inline src/librustc_borrowck/diagnostics.rs]: I'll try to convince him for another better sentence ! A bit of humor never kills ;)

**Comment by GuillaumeGomez** [comment]
Updated !

**Pull request merged**